### PR TITLE
[TD-2177]: release-5-lts: Update plugin compiler changes 

### PR DESCRIPTION
**Pull request by asutosh**
## Description
Revert the changes made to plugin compiler during
an earlier template sync.
This PR also fixes some issues in the new plugin compiler build workflow.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5428/merge
Commit:  
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5889733323)

File: ci/goreleaser/goreleaser-el7.yml [hunk 1]
```diff
@@ -14,6 +14,9 @@ builds:
       - -X github.com/TykTechnologies/tyk/gateway.VERSION={{.Version}} -X github.com/TykTechnologies/tyk/gateway.Commit={{.FullCommit}} -X github.com/TykTechnologies/tyk/gateway.buildDate={{.Date}} -X github.com/TykTechnologies/tyk/gateway.builtBy=goreleaser
     goos:
       - linux
+    env:
+      - CGO_ENABLED=1
+      - GO111MODULE=off
```
Comment (alephnull) [inline ci/goreleaser/goreleaser-el7.yml:19]: This change needs to be in templates right? Better commit it right now.

Comment (asutosh) [reply ci/goreleaser/goreleaser-el7.yml:19]: Updated in https://github.com/TykTechnologies/gromit/pull/201 and synced from it in https://github.com/TykTechnologies/tyk/pull/5428/commits/9ef5d1f769bbc248a589d77a7413b41e59dfcb89

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5428/merge
Commit: 9ef5d1f769bbc248a589d77a7413b41e59dfcb89 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5900454112)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5428/merge
Commit: 28da0a67d160003742245bf42bd08d97810683e3 
Triggered by: pull_request (@asutosh)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5900471866)

**Review by alephnull — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5428/merge
Commit: 19ac0e4cdf1d2a4fcba1045f20cc017128c662e5 
Triggered by: pull_request (@alephnull)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6008500650)

**Pull request merged**